src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -391,7 +392,21 @@ protected IList<CustomAttribute> SweepCustomAttributes (ICustomAttributeProvider
 				SweepCustomAttributes (provider);
 		}
 
-		protected virtual void SweepMethods (Collection<MethodDefinition> methods)
+		void AddReflectedAttr (MethodDefinition method, AssemblyDefinition assembly)
```

Comment:
For #2, mono embedders using the LLVM AOT backend can expect to see code size reductions when using this attribute: https://github.com/mono/mono/pull/13697

Reply:
> For #2, mono embedders using the LLVM AOT backend can expect to see code size reductions when using this attribute: mono/mono#13697

That's good to know.

Currently, we don't use the LLVM AOT backend so for us it would only increase size.

I don't really care what the switch looks like.  Something like the following may kill two birds with one stone.
```
protected virtual GetMonoCodeReflectedAttributeCtor()
{
     return System.Type.GetType ("Mono.Codegen+Reflected")?.GetConstructor (Array.Empty<System.Type> ());
}

		void AddReflectedAttr (MethodDefinition method, AssemblyDefinition assembly)
		{
			// Public methods have non-visible call sites by default, this attribute doesn't
			// help us at all.
			//
			// See mono_aot_can_specialize in aot-compiler.c in mono
			if (!method.IsPrivate)
				return;

			var reflectionMethod = GetMonoCodeReflectedAttributeCtor();
			if (reflectionMethod == null)
				return;

			MethodReference methodRef = assembly.MainModule.Import (reflectionMethod);
			method.CustomAttributes.Add(new CustomAttribute (methodRef));
		}
```

We could override `GetMonoCodeReflectedAttributeCtor` to turn this behavior off.  And the null check would fix the tests that are failing with this PR when ran on windows against the .NET Framework assemblies.

Reply:
Or you could pull this logic out into it's own step.  Solving the slightly confusion situation of this being in the `SweepStep` and then giving us the ability to simply not register this new step.

Reply:
I moved this to the CodeRewriter step. Can definitely put it in it's own step instead.